Regression(267645@main) ~3MB increase in binary size

cdumez opened the pull request:
#### 4dacd4adab9471bc6dfb8ce2c0a090a03cce1d52
<pre>
Regression(267645@main) ~3MB increase in binary size
<a href="https://bugs.webkit.org/show_bug.cgi?id=264896">https://bugs.webkit.org/show_bug.cgi?id=264896</a>
<a href="https://rdar.apple.com/118490755">rdar://118490755</a>

Reviewed by Brent Fulgham.

Pass comparison operands by const reference instead of by value, as
recommended by the clang team. This should generate code more similar
to the one before 267645@main.

* Source/JavaScriptCore/assembler/AbstractMacroAssembler.h:
* Source/JavaScriptCore/b3/B3Origin.h:
* Source/JavaScriptCore/b3/B3Type.h:
* Source/JavaScriptCore/b3/air/AirTmp.h:
* Source/JavaScriptCore/bytecode/ArithProfile.h:
* Source/JavaScriptCore/bytecode/CallVariant.h:
* Source/JavaScriptCore/bytecode/CodeBlockHash.h:
* Source/JavaScriptCore/bytecode/VirtualRegister.h:
* Source/JavaScriptCore/debugger/DebuggerScope.h:
* Source/JavaScriptCore/dfg/DFGAbstractHeap.h:
* Source/JavaScriptCore/dfg/DFGAbstractValueClobberEpoch.h:
* Source/JavaScriptCore/dfg/DFGEdge.h:
* Source/JavaScriptCore/dfg/DFGEpoch.h:
* Source/JavaScriptCore/dfg/DFGNodeFlowProjection.h:
* Source/JavaScriptCore/dfg/DFGRegisteredStructure.h:
* Source/JavaScriptCore/heap/Allocator.h:
* Source/JavaScriptCore/interpreter/CallFrame.h:
* Source/JavaScriptCore/jit/GPRInfo.h:
* Source/JavaScriptCore/jit/Reg.h:
* Source/JavaScriptCore/parser/SourceCodeKey.h:
* Source/JavaScriptCore/parser/VariableEnvironment.h:
* Source/JavaScriptCore/profiler/ProfilerUID.h:
* Source/JavaScriptCore/runtime/GenericOffset.h:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/JSScope.h:
* Source/JavaScriptCore/runtime/PageCount.h:
* Source/JavaScriptCore/runtime/StructureID.h:
* Source/JavaScriptCore/runtime/StructureTransitionTable.h:

Canonical link: <a href="https://commits.webkit.org/270891@main">https://commits.webkit.org/270891@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f271244c27c57ceb7fc27e7e574c989175a9576e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26675 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5288 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27915 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28888 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24396 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7107 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2685 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24308 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26937 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4109 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22904 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3616 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3649 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23895 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29373 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/23240 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24323 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24295 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29920 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/25873 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3688 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1881 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27819 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5135 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/33326 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4160 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7205 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3453 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4038 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
Let's see how this goes! R=me

cdumez committed:
Regression(267645@main) ~3MB increase in binary size
https://bugs.webkit.org/show_bug.cgi?id=264896
rdar://118490755

Reviewed by Brent Fulgham.

Pass comparison operands by const reference instead of by value, as
recommended by the clang team. This should generate code more similar
to the one before 267645@main.

* Source/JavaScriptCore/assembler/AbstractMacroAssembler.h:
* Source/JavaScriptCore/b3/B3Origin.h:
* Source/JavaScriptCore/b3/B3Type.h:
* Source/JavaScriptCore/b3/air/AirTmp.h:
* Source/JavaScriptCore/bytecode/ArithProfile.h:
* Source/JavaScriptCore/bytecode/CallVariant.h:
* Source/JavaScriptCore/bytecode/CodeBlockHash.h:
* Source/JavaScriptCore/bytecode/VirtualRegister.h:
* Source/JavaScriptCore/debugger/DebuggerScope.h:
* Source/JavaScriptCore/dfg/DFGAbstractHeap.h:
* Source/JavaScriptCore/dfg/DFGAbstractValueClobberEpoch.h:
* Source/JavaScriptCore/dfg/DFGEdge.h:
* Source/JavaScriptCore/dfg/DFGEpoch.h:
* Source/JavaScriptCore/dfg/DFGNodeFlowProjection.h:
* Source/JavaScriptCore/dfg/DFGRegisteredStructure.h:
* Source/JavaScriptCore/heap/Allocator.h:
* Source/JavaScriptCore/interpreter/CallFrame.h:
* Source/JavaScriptCore/jit/GPRInfo.h:
* Source/JavaScriptCore/jit/Reg.h:
* Source/JavaScriptCore/parser/SourceCodeKey.h:
* Source/JavaScriptCore/parser/VariableEnvironment.h:
* Source/JavaScriptCore/profiler/ProfilerUID.h:
* Source/JavaScriptCore/runtime/GenericOffset.h:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/JSScope.h:
* Source/JavaScriptCore/runtime/PageCount.h:
* Source/JavaScriptCore/runtime/StructureID.h:
* Source/JavaScriptCore/runtime/StructureTransitionTable.h:

Canonical link: https://commits.webkit.org/270891@main
